ERROR: "TypeError: Path must be a string. Received undefined"

First, thank you for providing this package. I find it very useful for comparing CSS changes and grabbing screen shots for other documentation.

I had it all working with v2.0.3 but recently updated while i was playing with yarn.
I get the following error on v2.0.4, v2.0.5 & 2.0.6.

```
Running CasperJS with:  [ '[[ PATH ]]\\node_modules\\backstopjs\\capture\\genBitmaps.js' ]
COMMAND | Command `reference` ended with an error
COMMAND | TypeError: Path must be a string. Received undefined
              at assertPath (path.js:8:11)
              at Object.win32.join (path.js:221:5)
              at module.exports ([[ PATH ]]\node_modules\backstopjs\core\util\findExecutable.js:16:15)
              at module.exports ([[ PATH ]]\node_modules\backstopjs\core\util\runCasper.js:30:23)
              at [[ PATH ]]\node_modules\backstopjs\core\util\createBitmaps.js:51:23

npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\ProgramFiles\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "bjs:question:ref"
npm ERR! node v5.5.0
npm ERR! npm  v3.3.12
```

I've removed the path from the output ([[ PATH]]) to keep it simple. 

SETUP:
OS: Win10
NODE: 5.5.0
NPM: 3.3.12

--- a/core/util/findExecutable.js
+++ b/core/util/findExecutable.js
@@ -1,17 +1,18 @@
 var path = require('path');
 
-module.exports = function(module, bin) {
+module.exports = function (module, bin) {
 
-  if (module === 'phantomjs-prebuilt') {
-    return require('phantomjs-prebuilt').path;
+  try {
+    if (module === 'phantomjs-prebuilt') {
+      return require('phantomjs-prebuilt').path;
+    }
+
+    var pathToExecutableModulePackageJson = require.resolve(path.join(module, 'package.json'));
+    var executableModulePackageJson = require(pathToExecutableModulePackageJson);
+    var relativePathToExecutableBinary = executableModulePackageJson.bin[bin] || executableModulePackageJson.bin;
+    var pathToExecutableModule = pathToExecutableModulePackageJson.replace('package.json', '');
+    return path.join(pathToExecutableModule, relativePathToExecutableBinary);
+  } catch (e) {
+    throw new Error('Couldn\'t find executable for module "' + module + '" and bin "' + bin + '"\n' + e.message)
   }
-
-  // get the absolute path for package.json of a node_module
-  var packageJSON = require.resolve(path.join(module, 'package.json'));
-
-  // then get the executable name
-  var relativeBinary = require(packageJSON).bin[bin];
-
-  // return a path to the executable inside the execuatable's package
-  return path.join(packageJSON.replace('package.json', ''), relativeBinary);
 };

--- a/test/core/util/findExecutable_spec.js
+++ b/test/core/util/findExecutable_spec.js
@@ -1,0 +1,21 @@
+var mockery = require('mockery');
+var findExecutable = require('../../../core/util/findExecutable');
+var assert = require('assert');
+
+describe('findExecutable_spec.js', function () {
+  before(function () {
+    mockery.enable({useCleanCache:true});
+  });
+
+  it('should resolve the path to phantomjs-prebuilt correctly', function () {
+    mockery.registerMock('phantomjs-prebuilt', {path: 'path-to-phantomjs'});
+    var pathToPhantomjs = findExecutable('phantomjs-prebuilt');
+    assert.equal(pathToPhantomjs, 'path-to-phantomjs')
+  });
+
+  it('should print a meaningful error the executable cannot be found', function () {
+    assert.throws(function() {
+      findExecutable('no-installed-module', 'invalid-bin');
+    }, /Couldn't find executable for module "no-installed-module" and bin "invalid-bin"/);
+  });
+});

